### ISSUE TITLE
Any luck on OpenCL?

### ISSUE THREAD

**Issue by Dexdev08** [InonS/Open-GPGPU-ANN#1]
I actually wanted to utilize my Intel HD5000 GPU on my laptop (although the RAM is a bit small for this).  Any luck on your side?

I have tried Hugh's libraries - I'm a bit of a n00b so for keras, i'm using the newer version.

**Comment by InonS** [comment]
At this point I'm not even sure what I thought was successful even was...

Please see my conversation with Hugh
<https://github.com/hughperkins/tf-coriander/issues/60>. TL;DR it ends with
both of us giving up.

https://github.com/hughperkins/tf-coriander/issues/60

**Comment by Dexdev08** [comment]
I see! I really wish opencl was more developed in deep learning.  I wanted to use existing hardware.

At this point, i am decided to bite the bullet and just setup my own deep learning rig with an nvidia gpu.

**Comment by Dexdev08** [comment]
My experience is that my gpu in a laptop, even if i get tf corainder running on toy problems, will not be able to handle any practical problems.  Note that the ram is just 384mb for the gpu.

**Comment by hughperkins** [comment]
Yes, you'd never train on a laptop gpu. You might use it for development, but any library in mainstream usage, notably Caffe, Torch, pyTorch, Tensorflow, ... will let you develop on your cpu, and then train later on a GPU, once you've got it working on the cpu anyway.

Personally, I dont have access to any discrete AMD GPUs. There are none in the Cloud anywhere, that I've seen. That limits the extent to which I can use myself any OpenCL products I create :)

**Comment by InonS** [comment]
I was hoping to develop on my laptop AMD GPU, thinking it would be faster
than on my CPU. An iterative development process is essentially.
If the production model (implemrnted in OpenCL) could run on nVidia GPUs,
that would complete the cycle.
